From 6e87687304b14ff71729885ec9cfacee963fbd71 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Wed, 13 Feb 2019 17:52:00 +0200
Subject: [PATCH 420/741] arm64: dts: fix the LS104x QDS mdio-mux support

commit 789768c250d2a53d090acdda66c5379a3e457d40 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Set the "simple-bus" compatible and the fpga ranges in order to
successfully probe the mdio-mux-emi1.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1043a-qds.dts | 3 ++-
 arch/arm64/boot/dts/freescale/fsl-ls1046a-qds.dts | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds.dts
index ccba39c7a34a..447fd7f007aa 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds.dts
@@ -78,10 +78,11 @@
 	};
 
 	fpga: board-control@2,0 {
-		compatible = "fsl,ls1043aqds-fpga", "fsl,fpga-qixis";
+		compatible = "fsl,ls1043aqds-fpga", "fsl,fpga-qixis", "simple-bus";
 		reg = <0x2 0x0 0x0000100>;
 		#address-cells = <1>;
 		#size-cells = <1>;
+		ranges = <0 2 0 0x100>;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds.dts
index d9c06a4b0e96..d1a2ca51346c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds.dts
@@ -167,8 +167,9 @@
 	};
 
 	fpga: board-control@2,0 {
-		compatible = "fsl,ls1046aqds-fpga", "fsl,fpga-qixis";
+		compatible = "fsl,ls1046aqds-fpga", "fsl,fpga-qixis", "simple-bus";
 		reg = <0x2 0x0 0x0000100>;
+		ranges = <0 2 0 0x100>;
 	};
 };
 
-- 
2.17.1

